scripts/trace-context.pl

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/perl -w
+#
+## Copyright (C) 1996-2022 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+# Reads cache.log and displays lines that correspond to a given CodeContext.
+
+use strict;
+use warnings;
+
+die("usage: $0 <CodeContext id> [log filename]\n") unless @ARGV;
+my $ContextId = shift;
+
+my $GroupSeparator = "--\n";
```

Comment:
Disclaimer: Other similar tools in scripts/ use an empty line instead of this grep-like `--` separator. The grep-like separator is better because it allows the user to distinguish log lines from tool markings in more cases -- cache.log does have quite a few native empty lines... However, this improvement creates an inconsistency across the `scripts/trace-*` space.

I think it is best to remove that (minor) inconsistency using a separate PR. I do not have the time to do that immediately, do not want to forget offering this handy script, and do not want to offer an inferior implementation after discovering the "standard" `--` group separator. I am sure that the other scripts can use some polishing/updates in addition to the separator changes.

However, if others insist I will replace this separator with a new line:

```suggestion
my $GroupSeparator = "\n"; # TODO: use "--\n" instead, like grep does
```

Reply:
> empty lines should only exist within those

There are a few other empty line cases, but that problem does not matter here.